src/io/flutter/editor/PropertyEditorPanel.java

```diff
@@ -0,0 +1,840 @@
+  }
+
+  void maybeLoadProperties() {
+    updateWidgetDescription();
+  }
+
+  public void initalize(
+    DiagnosticsNode node,
+    EventStream<List<FlutterOutline>> currentOutlines,
+    EventStream<VirtualFile> activeFile
+  ) {
+    this.node = node;
+    this.activeFile = activeFile;
+    currentOutlines.listen(this::outlinesChanged, true);
+    if (showWidgetEditToolbar) {
+      final WidgetEditToolbar widgetEditToolbar =
+        new WidgetEditToolbar(true, currentOutlines, activeFile, project, flutterDartAnalysisService);
+      final ActionToolbar toolbar = widgetEditToolbar.getToolbar();
+      toolbar.setShowSeparatorTitles(true);
+      setToolbar(toolbar.getComponent());
+    }
+
+    rebuildUi();
+  }
+
+  protected void rebuildUi() {
+    // TODO(jacobr): be lazier about only rebuilding what changed.
+    final Component focusOwner = KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner();
+    if (focusOwner != null) {
+      if (isAncestorOf(focusOwner)) {
+        for (Map.Entry<String, JComponent> entry : fields.entrySet()) {
+          if (entry.getValue().isAncestorOf(focusOwner) || entry.getValue() == focusOwner) {
+            previouslyFocusedProperty = entry.getKey();
+            break;
+          }
+        }
+      }
+      else {
+        previouslyFocusedProperty = null;
+      }
+    }
+    removeAll();
+    // Layout Constraints
+    // Column constraints
+    final MigLayout manager = new MigLayout(
+      "insets 3", // Layout Constraints
+      fixedWidth ? "[::120]5[:20:400]" : "[::120]5[grow]", // Column constraints
+      "[23]4[23]"
+    );
+    setLayout(manager);
+    int added = 0;
+    for (FlutterWidgetProperty property : properties) {
+      final String name = property.getName();
+      if (name.startsWith("on") || name.endsWith("Callback")) {
+        continue;
+      }
+      if (name.equals("key")) {
+        continue;
+      }
+      if (name.equals("child") || name.equals("children")) {
+        continue;
+      }
+      if (name.equals("Container")) {
+        final List<FlutterWidgetProperty> containerProperties = property.getChildren();
+        // TODO(jacobr): add support for container properties.
+        continue;
+      }
+      // Text widget properties to demote.
+      if (name.equals("strutStyle") || name.equals("locale") || name.equals("semanticsLabel")) {
+        continue;
+      }
+      final String documentation = property.getDocumentation();
+      JComponent field = null;
+
+      if (property.getEditor() == null) {
+        // TODO(jacobr): detect color properties more robustly.
+        final boolean colorProperty = name.equals("color");
+        final String colorPropertyName = name;
+        if (colorProperty) {
+          field = buildColorProperty(name, property);
+        }
+        else {
+          String expression = property.getExpression();
+          /*if (expression == null || expression.isEmpty()) {
+            continue;
+          }*/
```

Comment:
Can this be either uncommented or removed?

Reply:
removed.